crates/mr_bundle/src/bundle.rs

```diff
@@ -59,206 +78,181 @@ where
-    /// default location of the generated Bundle file.
-    ///
-    /// This will only be different than the Manifest path itself if the
-    /// Manifest::path impl specifies a nested path.
-    ///
-    /// Will return None if the `path` does not actually end with the
-    /// manifest relative path, meaning that either the manifest file is
-    /// misplaced within the unpacked directory, or an incorrect path was
-    /// supplied.
-    #[cfg(feature = "packing")]
-    pub fn find_root_dir(&self, path: &Path) -> MrBundleResult<PathBuf> {
-        crate::util::prune_path(path.into(), M::path()).map_err(Into::into)
+        let err =
+            Bundle::new(manifest.clone(), vec![("1.thing".into(), vec![1].into())]).unwrap_err();
+        assert!(
+            matches!(err, MrBundleError::MissingResources(ref resources) if resources.contains(&"2.thing".into())),
+            "Got other error: {err:?}"
+        );
+
+        let err = Bundle::new(
+            manifest,
+            vec![
+                ("1.thing".into(), vec![1].into()),
+                ("2.thing".into(), vec![2].into()),
+                ("3.thing".into(), vec![3].into()),
+            ],
+        )
+        .unwrap_err();
+        assert!(
+            matches!(
+                err,
+                MrBundleError::UnusedResources(ref resources) if resources.contains(&"3.thing".into())
+            ),
+            "Got other error: {err:?}"
+        );
     }
-}
 
-/// A manifest bundled together, optionally, with the Resources that it describes.
-/// The manifest may be of any format. This is useful for deserializing a bundle of
-/// an outdated format, so that it may be modified to fit the supported format.
-#[derive(Debug, PartialEq, Eq, Deserialize)]
-pub struct RawBundle<M> {
-    /// The manifest describing the resources that compose this bundle.
-    #[serde(bound(deserialize = "M: DeserializeOwned"))]
-    pub manifest: M,
+    #[test]
+    fn round_trip_pack_unpack() {
+        let manifest = TestManifest(vec!["1.thing".into(), "2.thing".into()]);
 
-    /// The full or partial resource data. Each entry must correspond to one
-    /// of the Bundled Locations specified by the Manifest. Bundled Locations
-    /// are always relative paths (relative to the root_dir).
-    pub resources: ResourceMap,
-}
+        let bundle = Bundle::new(
+            manifest.clone(),
+            vec![
+                ("1.thing".into(), vec![1].into()),
+                ("2.thing".into(), vec![2].into()),
+            ],
+        )
+        .unwrap();
 
-impl<M: serde::de::DeserializeOwned> RawBundle<M> {
-    /// Load a Bundle into memory from a file
-    pub async fn read_from_file(path: &Path) -> MrBundleResult<Self> {
-        crate::decode(&ffs::read(path).await?)
+        let packed = bundle.pack().unwrap();
+        let unpacked = Bundle::unpack(packed.reader()).unwrap();
+
+        assert_eq!(bundle, unpacked);
     }
-}
 
-#[cfg(test)]
-mod tests {
```

Comment:
Maybe add tests demonstrating erroring on MissingResources and UnusedResources

Reply:
That is tested by `bundle_validation`